src/core/mac/mac.cpp

```diff
@@ -1093,25 +1043,37 @@ void Mac::SentFrame(otError aError)
                      sendFrame.ToInfoString(stringBuffer, sizeof(stringBuffer)));
         otDumpDebgMac(GetInstance(), "TX ERR", sendFrame.GetHeader(), 16);
 
-        if (!RadioSupportsRetries() &&
-            mTransmitAttempts < sendFrame.GetMaxTxAttempts())
+        if (!RadioSupportsRetries() && mTransmitAttempts < sendFrame.GetMaxTxAttempts())
         {
-            StartCsmaBackoff();
             mCounters.mTxRetry++;
+            StartCsmaBackoff();
             ExitNow();
         }
 
         OT_UNUSED_VARIABLE(stringBuffer);
-        break;
     }
 
-    default:
-        assert(false);
-        break;
+    // Get the destination address and find the corresponding neighbor (if any).
+
+    sendFrame.GetDstAddr(dstAddr);
+    neighbor = GetNetif().GetMle().GetNeighbor(dstAddr);
```

Comment:
Is `neighbor` only used to update the link info?  If so, move this inside the following if-statement?

Reply:
It's currently only used in this `if` but I actually moved `neighbor` variable to whole method scope intentionally to have it be ready for later changes, e.g., adding other statistics (keeping track of number of tx attempts per frame per neighbor, etc.).